play/dbt-materialize/dbt/adapters/materialize/connections.py

```diff
@@ -0,0 +1,62 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+from dbt.adapters.postgres import PostgresConnectionManager
+from dbt.adapters.postgres import PostgresCredentials
+
+import dbt.exceptions
+from dataclasses import dataclass
+from dbt import flags
+from dbt.logger import GLOBAL_LOGGER as logger
+
+
+@dataclass
+class MaterializeCredentials(PostgresCredentials):
+    @property
+    def type(self):
+        return "materialize"
+
+
+class MaterializeConnectionManager(PostgresConnectionManager):
+    TYPE = "materialize"
+
+    @classmethod
+    def open(cls, connection):
+        connection = super().open(connection)
+        # Prevents psycopg connection from automatically opening transactions
+        # More info: https://www.psycopg.org/docs/usage.html#transactions-control
+        connection.handle.autocommit = True
+        return connection
+
+    def commit(self):
+        connection = self.get_thread_connection()
+        if flags.STRICT_MODE:
+            if not isinstance(connection, Connection):
+                raise dbt.exceptions.CompilerException(
+                    f"In commit, got {connection} - not a Connection!"
+                )
```

Comment:
What does STRICT_MODE mean in this context? This check seems strange to me.

Reply:
STRICT_MODE is a bit of a hidden dbt flag, but I believe it means that things typically raised as warnings will instead raise errors: https://github.com/fishtown-analytics/dbt/issues/1279